molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/filter/ClinvarMultiAllelicResultFilter.java

```diff
@@ -55,20 +48,32 @@ public ClinvarMultiAllelicResultFilter(List<AttributeMetaData> attributes)
 				for (int i = 0; i < clnSigs.length; i++)
 				{
 					int significantAlleleIndex = Integer.parseInt(clnAll[i]);
-
+					
 					// this means the no allele is associated with the gene of interest
 					if (significantAlleleIndex == -1) continue;
-
+					
 					// this means the allele is based on the reference
 					else if (significantAlleleIndex == 0)
 					{
+					
+						String resultRefAllele = entity.getString(VcfRepository.REF);
 						String refAllele = annotatedEntity.getString(VcfRepository.REF);
-
-						for (String annotatedEntityAltAllele : annotatedEntityAltAlleles)
+						for(int j = 0; j > annotatedEntityAltAlleles.length; j++ )
 						{
 							// if annotated entity allele equals the clinvar significant allele we want it!
-							if (refAllele.equals(annotatedEntityAltAllele))
+							if (refAllele.equals(resultRefAllele))
 							{
+								// if more than one clinsigs are available pair the right one with each allele
+								if (clnSigs[i].length() > 1)
```

Comment:
Code duplication?
Maybe you can create one generic method to merge these two chunks of code